config_src/external/GFDL_ocean_BGC/MOM_generic_tracer.F90

```diff
@@ -13,918 +13,267 @@ module MOM_generic_tracer
+  type(MOM_restart_CS), pointer :: restart_CSp => NULL() !< Restart control structure
+  type(ocean_OBC_type), pointer :: OBC => NULL() !<open boundary condition type
+  !> Pointer to the first element of the linked list of generic tracers.
+  !type(g_tracer_type), pointer :: g_tracer_list => NULL()
+
+end type MOM_generic_tracer_CS
 
 contains
 
-  !> Initializes the generic tracer packages and adds their tracers to the list
-  !! Adds the tracers in the list of generic tracers to the set of MOM tracers (i.e., MOM-register them)
-  !! Register these tracers for restart
-  function register_MOM_generic_tracer(HI, GV, param_file, CS, tr_Reg, restart_CS)
-    type(hor_index_type),       intent(in)   :: HI         !< Horizontal index ranges
-    type(verticalGrid_type),    intent(in)   :: GV         !< The ocean's vertical grid structure
-    type(param_file_type),      intent(in)   :: param_file !< A structure to parse for run-time parameters
-    type(MOM_generic_tracer_CS), pointer     :: CS         !< Pointer to the control structure for this module
-    type(tracer_registry_type), pointer      :: tr_Reg     !< Pointer to the control structure for the tracer
-                                                           !! advection and diffusion module.
-    type(MOM_restart_CS), target, intent(inout)  :: restart_CS !< MOM restart control struct
-
-    ! Local variables
-    logical :: register_MOM_generic_tracer
-    logical :: obc_has
-    ! This include declares and sets the variable "version".
-#   include "version_variable.h"
-
-    character(len=128), parameter :: sub_name = 'register_MOM_generic_tracer'
-    character(len=200) :: inputdir ! The directory where NetCDF input files are.
-    ! These can be overridden later in via the field manager?
-
-    integer :: ntau, axes(3)
-    type(g_tracer_type), pointer      :: g_tracer, g_tracer_next
-    character(len=fm_string_len)      :: g_tracer_name, longname,units
-    character(len=fm_string_len)      :: obc_src_file_name, obc_src_field_name
-    real :: lfac_in   ! Multiplicative factor used in setting the tracer-specific inverse length
-                      ! scales associated with inflowing tracer reservoirs at OBCs [nondim]
-    real :: lfac_out  ! Multiplicative factor used in setting the tracer-specific inverse length
-                      ! scales associated with outflowing tracer reservoirs at OBCs [nondim]
-    real, dimension(:,:,:,:), pointer   :: tr_field ! A pointer to a generic tracer field, in concentration units [conc]
-    real, dimension(:,:,:), pointer     :: tr_ptr   ! A pointer to a generic tracer field, in concentration units [conc]
-    real,    dimension(SZI_(HI),SZJ_(HI),SZK_(GV)) :: grid_tmask ! A 3-d copy of G%mask2dT [nondim]
-    integer, dimension(SZI_(HI),SZJ_(HI))          :: grid_kmt   ! A 2-d array of nk
-
-    register_MOM_generic_tracer = .false.
-    if (associated(CS)) then
```

Comment:
Please add something like
```
  call MOM_error(FATAL, "register_MOM_generic_tracer should not be called with the stub code "//&
         "in MOM6/config_src/external, as it does nothing.  Recompile using the full MOM_generic_tracer package.")
```

at this point so that people will not inadvertently do runs that should be using MOM generic tracers but aren't.